### PR TITLE
[stdlib] Some fixes for scalar names

### PR DESCRIPTION
**Pull request by Azoy**
There was a mistake with one of the CJK name ranges, and the first and last defined Unicode names (excluding variation selector) weren't being handled correctly.

### CODE REVIEW

**Comment by Azoy** [comment]
@swift-ci please smoke test

File: stdlib/public/stubs/Unicode/UnicodeScalarProps.cpp [hunk 1]
```diff
@@ -233,18 +233,30 @@ __swift_intptr_t _swift_stdlib_getScalarName(__swift_uint32_t scalar,
...
+      nextScalarOffset = _swift_stdlib_names_scalars[scalarIndex + i];
+      i += 1;
+    }
+  } else {
+    // This is the last element in the array which represents the last scalar
+    // name that Unicode defines (excluding variation selectors). This is
+    // U+E007F at the moment whose name is 'CANCEL TAG'. We add 4 because
+    // 'CANCEL' is found outside the byte area for words, but tag does.
+    // The resulting word indices look like this: [..., 0xFF, _, _, 0xXX];
+    nextScalarOffset = scalarOffset + 4;
```
Comment (milseman) [inline stdlib/public/stubs/Unicode/UnicodeScalarProps.cpp:259]: Is this code that will need to be updated if Unicode defines a name for a later scalar?

**Review by milseman — COMMENTED** [review]
Hooray for [better testing](https://github.com/apple/swift/pull/40181)

File: stdlib/public/stubs/Unicode/UnicodeScalarProps.cpp (hunk 1 above)
Comment (Azoy) [reply stdlib/public/stubs/Unicode/UnicodeScalarProps.cpp]: Yes, which is why I just pushed a commit that defines a macro for the last offset. 😅

**Comment by Azoy** [comment]
@swift-ci please smoke test

**Pull request merged**